iothub/device/src/Authentication/IotHubConnectionInfo.cs

```diff
@@ -3,15 +3,16 @@
 
 using System;
 using System.Diagnostics;
+using System.Threading.Tasks;
 using Microsoft.Azure.Devices.Client.Extensions;
 
 namespace Microsoft.Azure.Devices.Client
 {
-    internal sealed partial class IotHubConnectionString : IAuthorizationProvider
+    internal class IotHubConnectionInfo : IAuthorizationProvider
```

Comment:
I think the name was misleading. It isn't a connection string, rather it parses them, holds the info, produces one, and generates new passwords based on the auth information it was provided.

Also, it was split into 2 files, for no apparent reason. I consolidated them.